include/output.hxx

```diff
@@ -140,21 +147,22 @@ public:
 class ConditionalOutput : public Output {
 public:
   /// @param[in] base    The Output object which will be written to if enabled
-  ConditionalOutput(Output *base) : base(base), enabled(true), base_is_cond(false) {};
+  ConditionalOutput(Output *base) : base(base), enabled(true) {};
 
   /// Constuctor taking ConditionalOutput. This allows several layers of conditions
   /// 
   /// @param[in] base    A ConditionalOutput which will be written to if enabled
   /// 
   ConditionalOutput(ConditionalOutput *base)
-      : base(base), enabled(base->enabled), base_is_cond(true) {};
+      : base(base), enabled(base->enabled) {};
 
   /// If enabled, writes a string using C printf formatting
   /// by calling base->vwrite
   /// This string is then sent to log file and stdout (on processor 0)
   void write(const char *str, ...) override;
   void vwrite(const char *str, va_list va) override {
     if (enabled) {
+      ASSERT0(base != nullptr);
```

Comment:
Should be ASSERT1 or ASSERT2, as this is never set to a nullptr.
In the Fields we avoid the checks, and in the case of the fields there is actually a fair chance that they have not allocated the data ...
I guess have more well defined definitions of what the levels mean would be nice.
I think level0 should be only some very basic checks, that are fast and also have rather nasty consequences - and nullptr dereference results in a _clean_ segfault ...